asset/asset.go

```diff
@@ -801,20 +798,105 @@ var _ GenesisSigner = (*RawKeyGenesisSigner)(nil)
 
 // DeriveGroupKey derives an asset's group key based on an internal public
 // key descriptor, the original group asset genesis, and the asset's genesis.
-func DeriveGroupKey(genSigner GenesisSigner, rawKey keychain.KeyDescriptor,
-	initialGen Genesis, currentGen *Genesis) (*GroupKey, error) {
+func DeriveGroupKey(genSigner GenesisSigner, groupKey *GroupKey,
+	scriptLeaf *psbt.TaprootTapLeafScript, initialGen Genesis,
```

Comment:
I think we can use the `AssetGroup` struct in lieu of the group key, then the initial genesis?

Reply:
Yeah the `GroupKey` was intended to carry an optional tapscript root, but if that's no longer needed then `AssetGroup` does make more sense.

Reply:
GroupKey was dropped so i think the `DeriveGroupKey` args are ok as-is.